### ISSUE TITLE
Chaining `bind` and `bind_rustls`/`bind_openssl` on the same port.

### ISSUE THREAD

**Issue by daxpedda** [actix/actix-web#1199]
With @fafhrd91's help I figured out how to chain `HttpService`s with the goal to upgrade connections from `http` to `https` by redirection.

I was thinking it might be very useful to implement this on a high-level e.g. `actix-web`. Would that be acceptable or is this out of scope? I am willing to implement this myself.

**Comment by fafhrd91** [comment]
I don’t think this needs to be part of actix-web, it also can be third party component.

**Comment by fafhrd91** [comment]
I just want to tell, if you need this feature you have to build it, because nobody will. I;d like to avoid listing of dead feature requests.

**Comment by daxpedda** [comment]
> if you need this feature you have to build it, because nobody will.

This wasn't really a request for you to do it for me, I was asking if you would accept a PR from me.
But I totally undestand if this is out of scope!

**Comment by cdbattags** [comment]
It sounds like per @fafhrd91's comments he would accept a PR though, no?

@fafhrd91 most likely doesn't have capacity/time to do such a feature I'm guessing.

Either way would be nice if this was finished to be under the official actix repos, no?

**Comment by daxpedda** [comment]
Ah, I misunderstood then.

Would love to work on this, but considering the level of difficulty and that I'm completely new to actix and it's codebase, I will need some help for sure.

I will take another look and try to come up with a plan on how to integrate this and post it here for further discussion.

**Comment by fafhrd91** [comment]
i think this ticket should be moved to actix-net
